app/javascript/gobierto_data/lib/commons.js

```diff
@@ -168,3 +170,29 @@ export const sqlKeywords = [
   }
 ]
 
+export const convertVizToImgMixin = {
+  methods: {
+    data() {
+      return {
+        imageApi: null
+      }
+    },
+    convertVizToImg(opts) {
+      let node = document.querySelector('.gobierto-data-visualization--aspect-ratio-16-9');
+      const perspectiveChart = document.querySelector("perspective-viewer").shadowRoot
+      const perspectiveSidePanel = perspectiveChart.getElementById("side_panel")
+      const perspectiveTopPanel = perspectiveChart.getElementById("top_panel")
+      perspectiveSidePanel.style.display = "none"
+      perspectiveTopPanel.style.display = "none"
```

Comment:
🚨 Bad practices alert:

1. You're using classes to get the elements, binding a mixin to the template markup.
2. You're using querySelector when you have vue $refs
3. Why has the mixin to know about perspective or the nodes? That's should be done in the class who initializates the mixin.

Easiest way to proceed, would be passing the node you wanna convert as a function parameter:

```
convertVizToImg(node, opts) {
  // only do what the function says to do
  htmlToImage.toPng(node)....
```

Reply:
Done!